build/testing/config.go

```diff
@@ -0,0 +1,20 @@
+package testing
+
+import (
+	"context"
+
+	"dagger.io/dagger"
+)
+
+func Config(ctx context.Context, client *dagger.Client, container *dagger.Container) error {
+	_, err := client.Container().
+		Pipeline("validate advanced.yml with CUE").
+		From("golang:1.20-alpine3.16").
+		WithExec([]string{"go", "install", "cuelang.org/go/cmd/cue@v0.5.0"}).
+		WithFile("/flipt.schema.cue", container.File("config/flipt.schema.cue")).
+		WithFile("/config.yml", container.File("internal/config/testdata/advanced.yml")).
```

Comment:
is it possible to get the the default config output to yaml, then use that as the input here? reason being is this way still requires us to remember to update `advanced.yml` I think with new properties.

in an ideal situation we could depend on what gets written out from the Default config struct so that any time we add new fields without updating the accompanying `cue` or `jsonschema` files then this test should fail and remind us to update them

Reply:
alternatively, could we make this a unit test  in our existing `config_test.go` instead of an integration test and call `cue` from within Go code?

Reply:
Let me attempt to bake it into a unit test instead.
I will need to pull it a jsonschema Go implementation to support both kinds.
This looks promising https://github.com/xeipuuv/gojsonschema

I get what you're saying with the generate the yaml from the Go structures.
I will see what I can do.

Another thing we can do is actually generate the CUE from the Go structs themselves.

Reply:
Maybe not https://github.com/cue-lang/cue/issues/2217